src/hooks/useIsValidExecution.ts

```diff
@@ -22,14 +24,11 @@ const isContractError = (error: EthersError) => {
 
 // Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
 // This is ONLY used to check the validity of a transaction in `useIsValidExecution`
-const getPatchedSignerProvider = (
-  wallet: ConnectedWallet,
-  chainId: SafeInfo['chainId'],
-  readOnlyProvider: JsonRpcProvider,
-) => {
+const getPatchedSignerProvider = (wallet: ConnectedWallet, chain: ChainInfo, customRpc?: Record<string, string>) => {
```

Comment:
We might be able to remove this. What do you think @usame-algan?

Reply:
True, we should test if the validity check still works when in the tx flow while connected to a different network if we use `useWeb3` instead of this function.

Reply:
It looks like we can't, sadly. The contract instance requires a non-"read only" provider but it's possible that it can be one.